feat: Validate and Update Child Company Accounts on Rename/Adding Account

marination opened the pull request:
## **Issues:**

### **Rename**
- Assume parent company **Zoot**, with child companies **Child**, **Child2** and **Child3**
- Assume **Zoot** has a group account **Temporary Expenses** under parent account **Expenses**. The same get's added to all child company COAs as well
- If group account **Temporary Expenses** get's renamed or an account number get's added to it in the company **Zoot**, **_the same is not updated in Child Company Accounts_**. 

After renaming lets assume, it becomes **Temporary Expenditure**

### **Adding Account**
- Now If you add yet another account under account **Temporary Expenditure** in company **Zoot**, the same is account is not added in the child companies as account **Temporary Expenditure** does not exist in them, **Temporary Expenses** does

## **Fixes**

### **Rename**
- On renaming or changing account number in Parent Company Account, make sure the same is changed in Child Company corresponding accounts
 ![account-tree](https://user-images.githubusercontent.com/25857446/91724124-75249e80-ebba-11ea-8f5a-67faccc99e00.gif)

- On renaming account name/number in child company, check if `allow_account_creation_against_child_company` is disabled. If yes, check if account being renamed exists in parent company. If yes:
  ![Screenshot 2020-09-23 at 2 40 16 PM](https://user-images.githubusercontent.com/25857446/93993126-fd5a2600-fdab-11ea-8507-eeed3d82d808.png)

marination committed:
fix: Validate Missing Accounts in Child Companies

- If parent account exists in Parent and now child, throw error

marination committed:
fix: Rename/Update Child company Accounts as well

marination committed:
chore: Tests for Rename and Add child Account

marination committed:
fix: Redundant imports

deepeshgarg007 commented on erpnext/accounts/doctype/account/test_account.py:
```diff
@@ -111,6 +111,56 @@ def test_account_sync(self):
+
+		acc_tc_5 = frappe.db.get_value('Account', {'account_name': "Test Group Account", "company": "_Test Company 5"})
+		# Rename group account in one child company
+		update_account_number(acc_tc_5, "Test Modified Account")
```
Shouldn't we check while renaming the account in child company itself instead of throwing an error on addition of account in parent company? If same account in parent company exits then allow rename only via parent company.

If the above is not possible then I don't think we should throw a validation error in case account of same name in child company doesn't exist. We can just `msgprint` to notify the user instead of throwing a validation error.

marination committed:
fix: Check if account exists in parent company on rename

- Check if child company is dependent on parent company
- Check if account-to-be-renamed exists in parent, if yes, block
- Revised Test basd on the same

marination commented:
@deepeshgarg007 changes done, please check and let me know

marination committed:
Merge branch 'develop' into account-add-rename
